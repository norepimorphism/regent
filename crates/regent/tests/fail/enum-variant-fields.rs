fn main() {
    // One variant with one tuple field.
    #[regent::bitwise]
    enum A { _0(u32) }
    // One variant with two tuple fields.
    #[regent::bitwise]
    enum B { _0(u32, u32) }
    // One variant with one struct field.
    #[regent::bitwise]
    enum C { _0 { a: u32 } }
    // One variant with two struct fields.
    #[regent::bitwise]
    enum D { _0 { a: u32, b: u32 } }
    // One variant with fields, one field-less variant.
    #[regent::bitwise]
    enum E { _0(u32), _1 }
    // One field-less variant, one variant with fields.
    #[regent::bitwise]
    enum F { _0, _1(u32) }
    // Two variants, both with fields.
    #[regent::bitwise]
    enum G { _0(u32), _1(u32) }
}
