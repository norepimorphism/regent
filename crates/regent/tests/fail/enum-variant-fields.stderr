error: variant fields are not supported
 --> tests/fail/enum-variant-fields.rs:4:16
  |
4 |     enum A { _0(u32) }
  |                ^^^^^

error: variant fields are not supported
 --> tests/fail/enum-variant-fields.rs:7:16
  |
7 |     enum B { _0(u32, u32) }
  |                ^^^^^^^^^^

error: variant fields are not supported
  --> tests/fail/enum-variant-fields.rs:10:17
   |
10 |     enum C { _0 { a: u32 } }
   |                 ^^^^^^^^^^

error: variant fields are not supported
  --> tests/fail/enum-variant-fields.rs:13:17
   |
13 |     enum D { _0 { a: u32, b: u32 } }
   |                 ^^^^^^^^^^^^^^^^^^

error: variant fields are not supported
  --> tests/fail/enum-variant-fields.rs:16:16
   |
16 |     enum E { _0(u32), _1 }
   |                ^^^^^

error: variant fields are not supported
  --> tests/fail/enum-variant-fields.rs:19:20
   |
19 |     enum F { _0, _1(u32) }
   |                    ^^^^^

error: variant fields are not supported
  --> tests/fail/enum-variant-fields.rs:22:16
   |
22 |     enum G { _0(u32), _1(u32) }
   |                ^^^^^
